engine/classes/ElggVolatileMetadataCache.php

```diff
@@ -289,7 +289,7 @@ public function populateFromEntities($guids) {
 				"JOIN {$db_prefix}metastrings n ON n_table.name_id = n.id",
 			),
 			'selects' => array('n.string AS name', 'v.string AS value'),
-			'order_by' => 'n_table.entity_guid, n_table.time_created ASC',
+			'order_by' => 'n_table.entity_guid, n_table.id ASC',
```

Comment:
You can alter neither the `id` nor the `time_created` columns via API. If you recreate the metadata, e.g., `$entity->foo = [1, 2]` you're going to get new rows anyway. If I'm missing something, what series of events are you worried about?